foreshadow/smart/all.py

```diff
@@ -168,22 +174,33 @@ def pick_transformer(self, X, y=None, **fit_params):
             return_df=True, use_cat_names=True, handle_unknown="ignore"
         )
 
+        final_pipeline = SerializablePipeline(
+            [("fill_na", NaNFiller(fill_value="NaN"))]
+        )
+
         if self.y_var:
             return LabelEncoder()
         elif delim_diff < 0:
             delim = delimeters[delim_count.index(min(delim_count))]
-            return DummyEncoder(delimeter=delim)
+            final_pipeline.steps.append(
+                ("dummy_encodeer", DummyEncoder(delimeter=delim))
+            )
         elif unique_count <= self.unique_num_cutoff:
-            return ohe
+            final_pipeline.steps.append(("one_hot_encoder", ohe))
         elif (reduce_count <= self.unique_num_cutoff) and will_reduce:
```

Comment:
Check whether it should <= or > in the condition.

Reply:
It turns out that we still need to use `<=` (or more precisely, `<`). Here the `reduce_count` means the number of categories if we apply the `uncommon_remover`. Therefore, if the `potentially_reduced_count < self.unique_num_cutoff and will_reduce`, then we should apply the `uncommon_remover` in the final pipeline. The naming of the variable and the method `will_transform` are confusing and implicit. I'm changing it to something clearer.